sdk/src/main/java/com/horizen/account/state/Message.java

```diff
@@ -24,16 +24,16 @@ public class Message {
     private final boolean fakeMsg;
 
     public Message(
-            Optional<AddressProposition> from,
-            Optional<AddressProposition> to,
-            BigInteger gasPrice,
-            BigInteger gasFeeCap,
-            BigInteger gasTipCap,
-            BigInteger gasLimit,
-            BigInteger value,
-            BigInteger nonce,
-            byte[] data,
-            boolean fakeMsg
+        Address from,
```

Comment:
let's discuss the type of from:
`Address` or `Optional<Address>` ?
If I remember that empty `from` can be a case that we need to care about. It is negative flow for sure.
@alsala if you remember, pls join.

Reply:
I think this case is only valid for Transactions, not for Messages.

When we create a Message we need to know what the from address will be - it cannot be processed without it. Currently this was very misleading, because if `msg.from` was `None` it would often be treated as `null`, which turns into the Zero-Address when passed to libevm. See the original implementation of `msg.getFromAddressBytes`. One part of the refactoring is to make this behavior explicit.

Reply:
Ok, seems clear to me.
Let's double check the place where we transform Tx to Message. In case signature check occurs ALWAYS (by design) before, then it seems fine that `Message.from` is of type `Address`.
But in case signature validation is done after `Message` obj construction it seems more natural to fail it because of `None`.
PS: I mean the case when the received/constructed Tx miss signature at all.

Reply:
The signature check is included in `EthereumTransaction.semanticValidity()`. We go from Tx to Message in `StateDbAccountStateView.applyTransaction`, just before we call `applyMessage`, so I think we're good.

There were some checks on the `from` address in the `ForgerStakeMsgProcessor` which I removed because they made no sense to me. Without these it is now possible to create a forger stake from funds at the zero-address - but only within a non-persistent  `eth_call` of course. Things like these are also possible on Ethereum. If you omit `from` during `eth_call` or `eth_estimateGas` it will default to the zero-address.

Reply:
has sense!